pypika/terms.py

```diff
@@ -1060,6 +1061,31 @@ def tables_(self):
         return self.term.tables_
 
 
+class CustomFunction():
+    def __init__(self, name, params=None):
+        self.name = name
+        self.params = params
+
+    def __call__(self, *args, **kwargs):
+        if not self.hasParams():
+            return Function(self.name, alias=kwargs.get('alias'))
+
+        if not self.isValidFunctionCall(*args):
```

Comment:
could you please use slug case for function names? That's the pep8 standard.

Reply:
should also prefix the names with an underscore `_` to indicate that they are not meant to be called from outside the class.. `_is_valid_function_call` and `_has_params`

Reply:
Done